frontend/src/metabase/visualizations/components/settings/ChartSettingInputNumeric.tsx

```diff
@@ -14,20 +16,25 @@ const ChartSettingInputNumeric = ({
   value,
   ...props
 }: ChartSettingInputProps) => {
-  const [internalValue, setInternalValue] = useState(value);
+  const [internalValue, setInternalValue] = useState(value?.toString() ?? "");
 
   return (
     <ChartSettingNumericInput
-      type="number"
+      type="text"
       {..._.omit(props, "onChangeSettings")}
-      error={!!internalValue && isNaN(internalValue)}
+      error={internalValue !== "" && isNaN(Number(internalValue))}
       value={internalValue}
       onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-        const num = parseFloat(e.target.value);
-        setInternalValue(num);
+        const everyCharValid = e.target.value
```

Comment:
A side effect of changing the `type` to `"text"` is that `"text"` inputs be default allow any character to be typed. Here we prevent that  with this validation logic.